File: ZadaniaCSharp/UnitTesting/WhoLikesItTests.cs
using System.Collections.Generic;

namespace UnitTesting;

public class WhoLikesItTests
{
    [Fact]
    public void WhoLikesIt_ReturnsCorrectText_ForNoLikes()
    {
        // Arrange
        string[] names = new string[] { };

        AssertAsync.CompletesIn(5, () =>
        {
            Program.WhoLikesIt(names);
        });

        // Act
        string result = Program.WhoLikesIt(names);

        // Assert
        Assert.Equal("nikt tego nie lubi", result);
    }

    [Fact]
    public void WhoLikesIt_ReturnsCorrectText_ForOneLike()
    {
        // Arrange
        string[] names = new string[] { "Peter" };

        AssertAsync.CompletesIn(5, () =>
        {
            Program.WhoLikesIt(names);
        });

        // Act
        string result = Program.WhoLikesIt(names);

        // Assert
        Assert.Equal("Peter lubi to", result);
    }

    [Fact]
    public void WhoLikesIt_ReturnsCorrectText_ForTwoLikes()
    {
        // Arrange
        string[] names = new string[] { "Jacob", "Alex" };

        AssertAsync.CompletesIn(5, () =>
        {
            Program.WhoLikesIt(names);
        });

        // Act
        string result = Program.WhoLikesIt(names);

        // Assert
        Assert.Equal("Jacob i Alex lubią to", result);
    }

    [Fact]
    public void WhoLikesIt_ReturnsCorrectText_ForThreeLikes()
    {
        // Arrange
        string[] names = new string[] { "Max", "John", "Mark" };

        AssertAsync.CompletesIn(5, () =>
        {
            Program.WhoLikesIt(names);
        });

        // Act
        string result = Program.WhoLikesIt(names);

        // Assert
        Assert.Equal("Max, John i Mark lubią to", result);
    }

    [Fact]
    public void WhoLikesIt_ReturnsCorrectText_ForFourLikes()
    {
        // Arrange
        string[] names = new string[] { "Alex", "Jacob", "Mark", "Max" };

        AssertAsync.CompletesIn(5, () =>
        {
            Program.WhoLikesIt(names);
        });

        // Act
        string result = Program.WhoLikesIt(names);

        // Assert
        Assert.Equal("Alex, Jacob i 2 inne osoby lubią to", result);
    }
}

File: ZadaniaJava/ZadaniaJava/UnitTesting/WhoLikesItTests.java
package ZadaniaJava.UnitTesting;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

import ZadaniaJava.Zadania.Main;

public class WhoLikesItTests {

    @Test(timeout = 5000)
    public void testEmptyArray() {
        String[] people = {};
        assertEquals("nikt tego nie lubi", Main.whoLikesIt(people));
    }

    @Test(timeout = 5000)
    public void testOnePerson() {
        String[] people = { "Peter" };
        assertEquals("Peter lubi to", Main.whoLikesIt(people));
    }

    @Test(timeout = 5000)
    public void testTwoPeople() {
        String[] people = { "Jacob", "Alex" };
        assertEquals("Jacob i Alex lubią to", Main.whoLikesIt(people));
    }

    @Test(timeout = 5000)
    public void testThreePeople() {
        String[] people = { "Max", "John", "Mark" };
        assertEquals("Max, John i Mark lubią to", Main.whoLikesIt(people));
    }

    @Test(timeout = 5000)
    public void testMoreThanThreePeople() {
        String[] people = { "Alex", "Jacob", "Mark", "Max" };
        assertEquals("Alex, Jacob i 2 inne osoby lubią to", Main.whoLikesIt(people));
    }
}
